### DISCUSSION TITLE
There is no reliable way to delete or hide old/renamed Workflows?

### DISCUSSION THREAD

**Discussion by pcornelissen** [community/community#63316] [Actions]
### Select Topic Area

Product Feedback

### Body

There is #26256 which is in state answered, which it clearly is not, as you can see on numerous answers that show that the answer doesn't work besides very simple cases.

Please unanswer #26256 or use this thread and provide a real solution, which also works for workflows with many runs and ideally without the need to create a script that is not portable across platforms.

To summarize #26256:

Workflows can't be removed or renamed without having old runs leading to the workflow still be shown in the sidebar. There is no "hide", "cleanup" or delete possibility for workflows as a whole.

Suggested solutions in the old thread: (they are suggested over and over again)

1. remove workflows by hand and it will disappear
2. Use a script to use the gh API to remove the runs

**But** that doesn't work in all cases where you have a significant number of runs!

People have trouble with the script, you run into rate limiting of the API, people are not savvy enough for the scripts and some even complained that it remained visible even after they removed all runs, but that may have been a bug that may or may not be fixed now.

So to really fix the problem you need to provide a functionality in the UI to remove a workflow including its runs and/or an API endpoint that provides that with a single call to avoid rate limiting.

The first would be the best solution, because not everyone who uses GitHub is fond of running scripts.

For all discussion participants, please refrain from "me too" comments, that just spams all our inboxes with useless notifications, use the emoticon-reactions to express support.

**Comment by timc13** [comment]
This comment from https://github.com/orgs/community/discussions/26256#discussioncomment-6673330 seems like a very reasonable ask.

I am going to paste the mockup here again:
![259223696-e84f20dd-f823-4abf-9e47-466eb32ed640](https://github.com/community/community/assets/74982/e4736b99-2dc1-40a9-a9e8-5a2980db1c3c)

**Reply by pcornelissen** [reply]
That would be a nice solution. Maybe someone from the team takes the time to have a look at it and maybe reads the attached old discussion to see how many people would like to see a solution for that. :-)

**Reply by vaidik** [reply]
👍  Its important to have a way to clean this up. We have a heavy dependence on Github Actions. As a result, we create new workflows for testing. Unknowingly we created many and now there is no way to clean it up. So the developer experience of finding the right workflow to run in a long list of workflows (where mostly are irrelevant) is pretty poor right now.

**Comment by Akash1134** [comment]
Hi @pcornelissen 

Thanks for sharing and we appreciate your patience. We have highlighted this internally with the product team.

While it has been a while since we have updated on the thread you've mentioned , the absence of an update is simply that - there has been no update because there is so far nothing to report. That doesn't mean it isn't being worked on - it might simply be that we're unable to share details about work that isn't ready for release.

As a GitHub user, the most you can do is to share your feedback on this Discussion space with the correct tags (include examples), and trust that we are always working to make GitHub as awesome as possible. We can't guarantee that every bit of feedback sees action, but we are paying attention.

**Reply by pcornelissen** [reply]
Hi Akash, thanks for your reply.
I understand that not every discussion or small seed, which might form a plan can be reported back to us users. It was just a very long time without real feedback, that makes us a bit anxious ;-)
Nevertheless, thanks for the heads up. We're looking forward to solutions that may come in the future.

**Comment by rpekarek-tc** [comment]
We are also looking for this functionality

**Comment by 3dfoster** [comment]
Also looking for this pretty basic functionality.

**Comment by andre68723** [comment]
no solution, but using this meanwhile as workaround https://github.com/Mattraks/delete-workflow-runs

**Comment by abdul-musabbir** [comment]
Number #2

**Comment by Seung-o** [comment]
I have no workflow-run but workflow is created.. Generally, as i know, if all runs are removed, workflow should be disappear. but it wasn't.

A created (garbage) workflow name is equal to file path ( e.g. .github/workflows/some-action.yml ). 

Can i fix it?

![스크린샷 2023-11-27 오후 4 17 34](https://github.com/community/community/assets/79891873/99d9cd5b-5287-480b-8f9a-9b299d86de6f)

**Reply by WardenDrew** [reply]
We too have this issue. We had an old workflow. All the runs have been deleted. The yml file has been deleted. And yet it persists in the UI

**Comment by ssbarnea** [comment]
There are lots of workflows that we do not want to make visible in the default view because they add noise, most common example are workflows that run on actions like issue update or relabeling.

They add a huge amount of noise and make hard to identify real build.

... if only a GHA  product manager would read that and act on it, because it should be quite easy to enable some kind of filtering, as this can be achieved on frontend side only and rely on something like workflow name maybe. Even a generic convention like hiding workflows if they start with `.`, like on unix filenames.

**Comment by jakeleveroni** [comment]
So much for the productivity boom AI was supposed to bring about. Simple common sense QoL update been dead in the water for years.
